runtime/kusama/src/xcm_config.rs

```diff
@@ -335,7 +341,7 @@ impl xcm_executor::Config for XcmConfig {
 	type SubscriptionService = XcmPallet;
 	type PalletInstancesInfo = AllPalletsWithSystem;
 	type MaxAssetsIntoHolding = MaxAssetsIntoHolding;
-	type FeeManager = ();
+	type FeeManager = XcmFeesToAccount<Self, SystemParachains, AccountId, TreasuryAccount>;
```

Comment:
with this change `() -> XcmFeesToAccount`
Is there any change or consequence for non-system parachains or end users?
I mean, is there anything which will start failing after this is upgraded on-live?
Something like: **non-system parachain should drip their sovereign account on relay chain**?
If so, maybe should be part of release notes or something like that.

Reply:
actual `impl FeeManager for ()` [here](https://github.com/paritytech/polkadot/blob/kckyeung/xcm-fee-manager/xcm/xcm-executor/src/traits/fee_manager.rs#L54-L57) behaves as everything is waived now,
so, iiuc [`charge_fees`](https://github.com/paritytech/polkadot/blob/kckyeung/xcm-fee-manager/xcm/xcm-executor/src/lib.rs#L242-L252) and [`takes_fee`](https://github.com/paritytech/polkadot/blob/kckyeung/xcm-fee-manager/xcm/xcm-executor/src/lib.rs#L930-L937) they never do `AssetTransactor::withdraw_asset`, but now with `XcmFeesToAccount` they will do `AssetTransactor::withdraw_asset`,

so I think this is the change, that origin needs to have some balance > ED + fees, but I dont know how much this is a really issue. I can imaging just some scenarios/calls from some non-system parachain that does not have sovereign account with balance on relay chain, so after this change they will need to drip some DOTs/KSMs to pass

Reply:
Right, so I just took a look at where we collect fees in the XCM executor: it looks like any instruction where it creates and sends an XCM back to the origin would require the origin to pay for fees, and this includes the `Query*` instructions, the `Report*` instructions, the `ExportMessage` instruction and most of the cross-chain asset transfer instructions (e.g. `InitiateTeleport`).

This isn't a huge deal, I think what we do need to do is to display delivery fees in the UI clearly so that users are informed about how much they need to pay, otherwise their XCM would fail to be sent.